templates/cart/cart-shipping.php

```diff
@@ -58,7 +58,11 @@
 			<?php endif; ?>
 			<?php
 		elseif ( ! $has_calculated_shipping || ! $formatted_destination ) :
-			echo wp_kses_post( apply_filters( 'woocommerce_shipping_may_be_available_html', __( 'Enter your address to view shipping options.', 'woocommerce' ) ) );
+			if ( is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' ) ) {
+				echo wp_kses_post( apply_filters( 'woocommerce_shipping_may_be_available_cart_nocalc_html', __( 'Shipping options will be updated at checkout.', 'woocommerce' ) ) );
```

Comment:
Please update the message to:

```
Shipping costs are calculated during checkout
```

Also this `woocommerce_shipping_may_be_available_cart_nocalc_html` filter could get a more descriptive name, like `woocommerce_shipping_on_cart_not_enabled_html`.

Reply:
I used **Shipping options ...** in line with #22467. Using **Shipping costs ...** would deviate from that messaging. 

If the current message is not suitable, then we may use 

> Shipping options are calculated during checkout



Wrt hook, `woocommerce_shipping_on_cart_not_enabled_html` would imply that shipping is not enabled on the cart.

However, here shipping is enabled, but the shipping calculator is not enabled, we can't detect shipping destination, and there aren't any shipping methods for this scenario set up by the shop manager. Thus, we have to wait until checkout to find the exact shipping destination. Then if we don't have any shipping methods for that we can say we absolutely have no shipping options for the customer.

Maybe `woocommerce_shipping_may_be_available_calc_not_enabled_html` would be better? This would also align with `woocommerce_shipping_may_be_available_html` hook which is right after it and is contextually related.

Also, maybe a rework on your suggestion `woocommerce_shipping_not_available_on_cart_html`.

Reply:
> Wrt hook, woocommerce_shipping_on_cart_not_enabled_html would imply that shipping is not enabled on the cart.

That filter will only run when `is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' )` returns true, so how shipping can be enabled and still hit this filter?

> If the current message is not suitable, then we may use

As I said, the message I suggested will fit better.

> is_cart() && 'no' === get_option( 'woocommerce_enable_shipping_calc' )

Reply:
Updated the message in the latest commit.

> That filter will only run when `is_cart()` && `'no' === get_option( 'woocommerce_enable_shipping_calc' )` returns true, so how shipping can be enabled and still hit this filter?

`'no' === get_option( 'woocommerce_enable_shipping_calc' )` checks if **_Shipping Calculator_** is enabled or not on the cart. 

So shipping can be enabled but we can't match any shipping methods automatically because either we can't locate shipping destination (Geolocation may be disabled) or matched zone does not have methods for current packages. But because we are on cart and shipping calculator is disabled, we can not be sure about shipping destination until the customer enters it manually on the checkout page.

Bottom line: Shipping is enabled but we neither have matching shipping methods available nor we have accurate shipping destination available to flat out say _No shipping available for %s_

Reply:
@dhavalgshah I still don't understand why we can't change the filter name as I requested. If shipping is disabled on cart, that `woocommerce_shipping_may_be_available_cart_nocalc_html` doesn't makes sense, because isn't "may be available", it's for sure unavailable, since `get_option( 'woocommerce_enable_shipping_calc' )` will returns `no`, so no shipping calculator will got rendered in the cart.

Let me know if you can update.